smart/characterization/dB_dt.py

```diff
@@ -0,0 +1,193 @@
+    sorted_labels : numpy.ndarray
+        An array where each unique label corresponds to a different feature on the solar disk.
+    dB_dt : Map
+        Map showcasing the change in magnetic field strength over time.
+    dt : Quantity
+        The time interval over which the change in magnetic field strength was measured.
+
+    Returns
+    -------
+    emergence_rates : list of Quantity
+        A list containing values for the flux emergence rate for each labelled feature in Gauss per second.
+    """
+    feature_masks = extract_features(sorted_labels)
+
+    emergence_rates = []
+    for feature_mask in feature_masks:
+        area_map, total_area = cosine_weighted_area_map(im_map, feature_mask)
+
+        magnetic_flux = np.nansum(dB_dt.data * area_map)
+
+        flux_emergence_rate = magnetic_flux / dt
+
+        emergence_rates.append(flux_emergence_rate)
+
+    return emergence_rates
+
+
+def get_properties(im_map, dB_dt, dt, sorted_labels):
+    """
+    Calculate various properties for each detected feature.
+
+    Parameters
+    ----------
+    im_map : Map
+        Processed SunPy magnetogram map.
+    dB_dt : Map
+        Map showcasing the change in magnetic field strength over time.
+    dt : Quantity
+        The time interval over which the change in magnetic field strength was measured.
+    sorted_labels : numpy.ndarray
+        An array where each unique label corresponds to a different feature on the solar disk.
+
+    Returns
+    -------
+    properties : list
+        A list containing properties related to each individual feature
+    """
+
+    feature_masks = extract_features(sorted_labels)
+
+    properties = []
+    for i, feature_mask in enumerate(feature_masks, start=1):
+        total_area, area_map = cosine_weighted_area_map(im_map, feature_mask)
+
+        dBdt_data = dB_dt.data * u.G
+
+        magnetic_flux = np.nansum(dBdt_data * area_map)
+        flux_emergence_rate = magnetic_flux / dt
+
+        B = im_map.data * feature_mask * u.G
+        B_mean = np.nanmean(B)
+        B_std = np.nanstd(B)
+        B_min = np.nanmin(B)
+        B_max = np.nanmax(B)
+
+        flux_pos = np.nansum(B[B > 0] * area_map[B > 0])
+        flux_neg = np.nansum(B[B < 0] * area_map[B < 0])
+        flux_uns = np.nansum(np.abs(B) * area_map)
+        flux_imb = (flux_pos - flux_neg) / (flux_pos + flux_neg)
+
+        properties.append(
+            {
+                "feature label": i,
+                "flux emergence rate": flux_emergence_rate.to(u.Wb / u.s),
+                "mean B": B_mean.to(u.G),
+                "std B": B_std.to(u.G),
+                "minimum B": B_min.to(u.G),
+                "maximum B": B_max.to(u.G),
+                "positive flux": flux_pos.to(u.Wb),
+                "negative flux": flux_neg.to(u.Wb),
+                "unsigned flux": flux_uns.to(u.Wb),
+                "flux imbalance": flux_imb,
+                "total area": total_area.to(u.m**2),
+            }
```

Comment:
Not sure about some of these particularly the unsigned flux have a look the idl code here

https://github.com/TCDSolar/SMART/blob/0530e19ba2aa7bc553153a97379460cae041e609/property_modules/ar_magprop.pro#L95-L113